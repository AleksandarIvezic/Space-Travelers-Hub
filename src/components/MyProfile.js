import React from 'react';
import { useSelector } from 'react-redux';

const MyProfile = () => {
  const rockets = useSelector((state) => state.rocketsReducer.rockets);
  let reservedRockets;
  if (rockets) {
    reservedRockets = rockets.filter((rocket) => rocket.reserved);
  }

  const missions = useSelector((state) => state.missions.missions)
  let reservedMissions;
  if (missions) {
    reservedMissions = missions.filter((mission) => mission.reserved)
  }

  return (
    <div className="d-flex w-75 mx-auto">
      <div className="my-missions w-50 me-2">
        <h4>My Missions</h4>
        {reservedMissions && (
          <ul className="list-group">
            {reservedMissions.map((reservedMission) => (
              <li key={reservedMission.mission_id} className="list-group-item">{reservedMission.mission_name}</li>
            ))}
          </ul>
        )}
      </div>

      <div className="my-rockets w-50 ms-2">
        <h4>My Rockets</h4>
        {reservedRockets && (
          <ul className="my-rockets list-group">
            {reservedRockets.map((rocket) => (
              <li
                key={rocket.id}
                className="list-group-item"
              >
                {rocket.name}
              </li>
            ))}
          </ul>
        )}
        {(!reservedRockets || !reservedRockets[0]) && (
          <h4 className="text-danger m-4">No reserved rockets! :(</h4>
        )}

      </div>
    </div>
  );
};

export default MyProfile;
