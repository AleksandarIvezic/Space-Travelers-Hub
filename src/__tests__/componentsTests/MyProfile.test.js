import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/storeConfig';
import MyProfile from '../../components/MyProfile';

describe('Test if MyProfile', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MyProfile />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
