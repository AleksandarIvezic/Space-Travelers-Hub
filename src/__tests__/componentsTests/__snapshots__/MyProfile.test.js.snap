// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if MyProfile renders correctly 1`] = `
<div
  className="d-flex w-75 mx-auto mt-4"
>
  <div
    className="my-missions w-50 me-2"
  >
    <h4>
      My Missions
    </h4>
    <h3
      className="text-danger"
    >
      Apparently you have not joined missions - yet
    </h3>
  </div>
  <div
    className="my-rockets w-50 ms-2"
  >
    <h4>
      My Rockets
    </h4>
    <h4
      className="text-danger m-4"
    >
      No reserved rockets! :(
    </h4>
  </div>
</div>
`;
